src/diffcalc_API/routes/ub.py

```diff
@@ -52,29 +56,43 @@ async def edit_reflection(
     store: HklCalcStore = Depends(get_store),
     collection: Optional[str] = Query(default=None, example="B07"),
 ):
+    if (params.retrieve_idx is None) and (params.retrieve_tag is None):
+        raise NoTagOrIdxProvidedError()
+
+    refl: str = (
+        f"index {params.retrieve_idx}"
+        if params.retrieve_idx is not None
+        else f"tag {params.retrieve_tag}"
+    )
     await service.edit_reflection(name, params, store, collection)
     return {
         "message": (
             f"reflection of crystal {name} in collection {collection} "
-            + f"with tag/index {params.tag_or_idx} edited. "
+            + "with %s edited" % refl
         )
     }
 
 
 @router.delete("/{name}/reflection")
 async def delete_reflection(
     name: str,
-    params: DeleteParams = Body(..., example={"tag_or_idx": "refl1"}),
+    tag: str = Query(default=None, example="refl1"),
+    idx: int = Query(default=None),
     store: HklCalcStore = Depends(get_store),
     collection: Optional[str] = Query(default=None, example="B07"),
 ):
+    if (idx is None) and (tag is None):
+        raise NoTagOrIdxProvidedError()
```

Comment:
We should probably also raise an error if both are set

Reply:
I've thought about this, and figured in the case where both are set I'll just take the index. But you're right, I think it's better to be clear that only one or the other should be set.